hpx/runtime/parcelset/parcelport_impl.hpp

```diff
@@ -723,6 +730,51 @@ namespace hpx { namespace parcelset
         }
 
         ///////////////////////////////////////////////////////////////////////
+        void send_parcel_immediate(locality const& locality_id, parcel p,
+            write_handler_type handler, std::shared_ptr<archive_type> const & archive,
+            new_gids_map && new_gids)
+        {
+            // If we are stopped already, discard pending parcels
+            if (hpx::is_stopped()) return;
+
+            error_code ec;
+            std::shared_ptr<connection> sender_connection =
+                connection_handler().create_connection(locality_id, ec);
+
+#if defined(HPX_DEBUG)
+            // verify the connection points to the right destination
+            HPX_ASSERT(locality_id == sender_connection->destination());
+            sender_connection->verify(locality_id);
+#endif
+
+#if defined(HPX_TRACK_STATE_OF_OUTGOING_TCP_CONNECTION)
+            sender_connection->set_state(parcelport_connection::state_send_pending);
+#endif
+            // encode the parcels
+            std::size_t num_parcels = encode_parcels(&p, 1,
+                    sender_connection->buffer_,
+                    archive_flags_,
+                    this->get_max_outbound_message_size(),
+                    &new_gids);
+
+            using hpx::parcelset::detail::call_for_each;
+            using hpx::util::placeholders::_1;
+            using hpx::util::placeholders::_2;
+            using hpx::util::placeholders::_3;
+            HPX_ASSERT(num_parcels == 1);
+            {
+                ++operations_in_flight_;
+
+                // send the parcel
+                sender_connection->async_write(
+                    util::bind(util::one_shot(handler),
+                        _1,  std::move(p)),
```

Comment:
@hkaiser I have replaced the CXX14_LAMBDAS with a bind(one shot) and move of the parce. It seems to work, but please tell me if this is not the correct way to do it.